src/main/webapp/ChapMap.js

```diff
@@ -475,4 +377,140 @@ class ChapMap {
 
     connection.send(JSON.stringify(jsonObject));
   }
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// MAP SHARING
+
+  /**
+   * @Private
+   * Sets the events related to sharing the map with another user
+   */
+  setMapShareEvents_() {
+    let shareFn = () => this.openSharePopup_();
+    this.addClickEvent_("shareBtnWrapper", shareFn);
+
+    let addEmailFn = () => this.addEmail_();
+    this.addClickEvent_("addEmail", addEmailFn);
+
+    let submitFn = () => this.submitSharing_();
+    this.addClickEvent_("share", submitFn);
+
+    let closeFn = () => this.closeSharePopup_();
+    this.addClickEvent_("close", closeFn);
+  }
+
+
+  /**
+   * @Private
+   * Sets a click-trigger event to the DOM element with the given id and
+   * sets the callback function to the one given
+   * @param {String} id the id of the element to be added
+   * @param {} fn the anonymous function to be called on click
+   */
+  addClickEvent_(id, fn) {
+    let btn = this.getEl_(id);
+    btn.addEventListener('click', fn);
+  }
+
+  /**
+   * @Private
+   * Opens the sharing popup and prevents the client from clicking on the map
+   */
+  openSharePopup_() {
+    let overlay = this.getEl_("share-popup");
+    overlay.classList.add("cover");
+
+    // let popup = this.getEl_("popup-window");
```

Comment:
Add a TODO above commented-out code or remove

Reply:
Whoops, I'll delete it.